### PR TITLE
split features in to various categories

### PR DESCRIPTION
**Pull request by ramaraochavali**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu WDYT about splitting features like this? If this is a good idea, I can fine tune and clean up the PR. Just posted to get your thoughts

**Comment by keithmattix** [comment]
+1 on this idea; much easier to reason about

File: pilot/pkg/features/xds.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import (
+	"time"
+
+	"go.uber.org/atomic"
+
+	"istio.io/istio/pkg/env"
+)
+
+// Define flags that affect XDS config generation here.
```
Comment (howardjohn) [inline pilot/pkg/features/xds.go:25]: Not clear the distinction between this and experimental

File: pilot/pkg/features/experimental.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	).Get()
+
+	StatusUpdateInterval = env.Register(
+		"PILOT_STATUS_UPDATE_INTERVAL",
+		500*time.Millisecond,
+		"Interval to update the XDS distribution status.",
+	).Get()
+
+	StatusQPS = env.Register(
+		"PILOT_STATUS_QPS",
```
Comment (howardjohn) [inline pilot/pkg/features/experimental.go:144]: tuning? But then PILOT_ENABLE_STATUS is not and its logical to group. Tricky to pick just one category!

**Review by howardjohn — COMMENTED** [review]
Overall lgtm

**Comment by zirain** [comment]
not a must, but any chance to help with https://github.com/istio/istio/issues/47925?

**Review by hzxuzhonghu — COMMENTED** [review]
I like this classify

File: pilot/pkg/features/xds.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/features/xds.go:25]: And these features should be in pilot

File: pilot/pkg/features/xds.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+	XDSCacheMaxSize = env.Register("PILOT_XDS_CACHE_SIZE", 60000,
+		"The maximum number of cache entries for the XDS cache.").Get()
+
+	XDSCacheIndexClearInterval = env.Register("PILOT_XDS_CACHE_INDEX_CLEAR_INTERVAL", 5*time.Second,
+		"The interval for xds cache index clearing.").Get()
+
+	XdsPushSendTimeout = env.Register(
+		"PILOT_XDS_SEND_TIMEOUT",
+		0*time.Second,
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/xds.go:110]: Do they belong to tune?

File: pilot/pkg/features/security.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		return res
+	}()
+
+	CertSignerDomain = env.Register("CERT_SIGNER_DOMAIN", "", "The cert signer domain info").Get()
+
+	UseCacertsForSelfSignedCA = env.Register("USE_CACERTS_FOR_SELF_SIGNED_CA", false,
+		"If enabled, istiod will use a secret named cacerts to store its self-signed istio-"+
+			"generated root certificate.").Get()
+
+	StackdriverAuditLog = env.Register("STACKDRIVER_AUDIT_LOG", false, ""+
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/security.go:97]: belong to telemetry

File: pilot/pkg/features/xds.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/features/xds.go:25]: I will move them to experimental for now and we can move to pilot once they are stable

File: pilot/pkg/features/xds.go (hunk 3 above)
Comment (ramaraochavali) [reply pilot/pkg/features/xds.go:110]: will move

File: pilot/pkg/features/experimental.go (hunk 2 above)
Comment (ramaraochavali) [reply pilot/pkg/features/experimental.go]: move these to tuning

**Comment by ramaraochavali** [comment]
> Only comment is that security is maybe a bit too broad/fuzzy of a category vis a vis the others and maybe authn/authz would be better.

I agree but left it with security for now and we can change later

**Review by bleggett — APPROVED** [review]
SGTM, as mentioned there will be some things where the classification will be a matter of opinion, but this is better than status quo.

**Review by keithmattix — APPROVED** [review]

**Comment by keithmattix** [comment]
Holding for folks to get their last reviews in

**Review by howardjohn — APPROVED** [review]

**Pull request merged**